### ISSUE TITLE
test exceptions not shown sometimes, "Error handling response" in repl

### ISSUE THREAD

**Issue by rutchkiwi** [cursive-ide/cursive#2994]
So I've got a test containing this: 

```
(deftest regulated-product-form
  (is (= 1 1))
  (let [stored-responses-for-banana (db/query db/db ["sdsfsdfsdf" "order-id" "banana-id"])]
    (is (= (count stored-responses-for-banana) 14))))
```

that db query is obviosly a syntax error.
When this is run using "run test under cursor" I get this in my REPL:

```
Error handling response - class java.lang.IndexOutOfBoundsException: Wrong line: -2. Available lines count: 642
Error handling response - class java.lang.IndexOutOfBoundsException: Wrong line: -2. Available lines count: 642
Error handling response - class java.lang.IndexOutOfBoundsException: Wrong line: -2. Available lines count: 642
Error handling response - class java.lang.IndexOutOfBoundsException: Wrong line: -2. Available lines count: 642
Error handling response - class java.lang.IndexOutOfBoundsException: Wrong line: -2. Available lines count: 642
```

And then, no exception is shown in the editor or REPL:
<img width="989" alt="Screenshot 2024-11-22 at 19 21 57" src="https://github.com/user-attachments/assets/8560e92a-0b48-40ae-9d8a-bd7024c72593">

which is annoying as I then need to run the test manually.

this however works fine:
```
(let [stored-responses-for-banana (db/query db/db ["sdsfsdfsdf"])]
    (is (= (count stored-responses-for-banana) 14)))
```

I'm on 1.14.0-2024.2.

**Comment by rutchkiwi** [comment]
keeps happening, I got exceptions thrown in intellij. Reported these, sending them in using the crash/bug reporter and referenced this issue

**Comment by cursive-ide** [comment]
I haven't managed to reproduce your case exactly, but I have put some guard rails around the failing check which should fix this. I suspect this is happening when a macro expansion returns a source line of -1.

**Comment by cursive-ide** [comment]
Fix released in 1.14.1-eap1 on 2024-12-11.
